File: server/model/networkstate.js
var mongoose = require('mongoose');

var networktypes = require('./networktypes.json');
var networkstateSchema = new mongoose.Schema({
  latitude: Number,
  longitude: Number,
  signalStrength: { type: Number, required: true },
  operatorName: String,
  networkProtocol: { type: String, enum: networktypes },
  date: { type: Date, default: Date.now }
});
mongoose.model('networkstate', networkstateSchema);

File: androidapp/app/src/main/java/eu/miaounyan/isthereanynetwork/model/NetworkState.kt
package eu.miaounyan.isthereanynetwork.model

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
data class NetworkState(
        @field:Json(name = "latitude") val latitude: Double,
        @field:Json(name = "longitude") val longitude: Double,
        @field:Json(name = "signalStrength") val signalStrength: Int,
        @field:Json(name = "operatorName") val operatorName: String,
        @field:Json(name = "date") val date: String, //FIXME: check if we can use a date type directly
        @field:Json(name = "networkProtocol") val networkProtocol: String
) : Parcelable